nbgrader/exchange/exchange.py

```diff
@@ -125,3 +125,43 @@ def start(self):
         self.init_src()
         self.init_dest()
         self.copy_files()
+
+    def get_current_user_courses(self):
+        """Check if student is enrolled in course"""
+        from jupyterhub.services.auth import HubOAuth
+        from tornado.httpclient import HTTPClient, HTTPRequest
+
+        if os.getenv('JUPYTERHUB_API_TOKEN'):
+            api_token = os.environ['JUPYTERHUB_API_TOKEN']
+        if not api_token:
+            self.exit("JUPYTERHUB_API_TOKEN env is required to run nbgrader. Did you launch it manually?")
+        hub_api_url = os.environ.get('JUPYTERHUB_API_URL') or 'http://127.0.0.1:8081/hub/api'
+        base_url = os.environ.get('JUPYTERHUB_SERVICE_PREFIX') or '/'
+        hub_prefix = base_url + "hub/"
+        self.hub_auth = HubOAuth(
+            parent=self,
+            api_token=api_token,
+            api_url=hub_api_url,
+            hub_prefix=hub_prefix,
+            base_url=base_url,
+        )
+        import json
+        # smoke check
+        if not self.hub_auth.oauth_client_id:
+            raise ValueError("Missing OAuth client ID")
+        auth_header = {
+                'Authorization': 'token %s' % api_token
+            }
+        client = HTTPClient()
+        req = HTTPRequest(url=hub_api_url + '/users/%s' % self.hub_auth.oauth_client_id.replace('user-', ''),
+            method='GET',
+            headers=auth_header,
+        )
+
+        resp = client.fetch(req)
+        groups = json.loads(resp.body.decode('utf8', 'replace'))['groups']
+        courses = set()
+        for group in groups:
+            if group.startswith('nbgrader-') or group.startswith('formgrade-'):
+                courses.add('-'.join(group.split('-')[1:]))
```

Comment:
here I assume that students are in the `nbgrader-course_id` Jupyterhub group and the instructors in `formgrade-course_id` group, is this good enough?

Reply:
You could do:

```python
courses.add(group.split('-', 1)[1])
```

to get everything after the first `-` without needing to re-join.